### DISCUSSION TITLE
Add extra file to define the video rom to use ?

### DISCUSSION THREAD

**Discussion by aram-gilbert** [markadev/AppleII-VGA#36] [Q&A]
Dears, 

  I'm using an Apple //e 65c02  with French ROM using the special french characters. I believe the VGA-Raspberry PI card has its own character set. 

In time, I was involved in WinApple settings and provided information on the side effect with screenshots. 

I was considering to update the application character set to load the character in the ROM at boot to get the localised version. But, from the threads, I've read that the fonts are stored in a memory accessible only from the video output circuit. 

It there a way to upload a 8KB character ROM from a file, the same way we update the firmware ? 

Video Roms collection : 

- [4K Collection](https://downloads.reactivemicro.com/Apple%20II%20Items/ROM_and_JEDEC/IIe/Video%20ROM/)

- [8K Collection](https://downloads.reactivemicro.com/Apple%20II%20Items/ROM_and_JEDEC/IIe/Video%20ROM/Dual-Euro/)

Reference : [https://github.com/AppleWin/AppleWin/issues/574](https://github.com/AppleWin/AppleWin/issues/574)

Thank you for your nice product. I rediscover my Apple //e in original condition with modern screen ! 

  Regards,

  Alexandre.

**Comment by markadev** [comment]
Hey there! Yes your understanding is correct.

You're in luck because the latest release of the firmware just [added many built-in character ROMs](https://github.com/markadev/AppleII-VGA/pull/31) that you can switch between!  You can use a `POKE` command or the [AppleIIVGA-Config.dsk](https://github.com/markadev/AppleII-VGA/releases/download/v1.2.0/AppleIIVGA-Config.dsk) to enable French-Canadian (which looks like an enhanced French ROM) and then you can save it to flash so it automatically gets selected at power-on.

I've read that Euro Apple II's have a toggle switch for switching between US/non-US but the VGA card won't be able to read that.  If you do need to switch back then you would use another `POKE` command or run the configuration disk.

There is also some support to load a custom character ROM but there's only [some registers](https://github.com/markadev/AppleII-VGA/blob/main/docs/Usage.md#registers-2--3---text-character-pattern-offset--data) so you would need to write your own program to load the data into those registers.

I found your AppleWin issue when I was searching for ROMs and it was definitely helpful to understand how the Euro video ROMs worked since I've never seen a real one, so thanks for that! 😃
